servicetalk-encoding-api/src/main/java/io/servicetalk/encoding/api/IdentityBufferEncoderDecoder.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright © 2021 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.encoding.api;
+
+import io.servicetalk.buffer.api.Buffer;
+import io.servicetalk.buffer.api.BufferAllocator;
+import io.servicetalk.concurrent.BlockingIterable;
+import io.servicetalk.concurrent.api.Publisher;
+import io.servicetalk.concurrent.internal.BlockingIterables;
+import io.servicetalk.oio.api.PayloadWriter;
+import io.servicetalk.serializer.api.Deserializer;
+import io.servicetalk.serializer.api.Serializer;
+import io.servicetalk.serializer.api.SerializerDeserializer;
+import io.servicetalk.serializer.api.StreamingDeserializer;
+import io.servicetalk.serializer.api.StreamingSerializer;
+import io.servicetalk.serializer.api.StreamingSerializerDeserializer;
+
+import static io.servicetalk.buffer.api.CharSequences.newAsciiString;
+
+final class IdentityBufferEncoderDecoder implements BufferEncoderDecoder {
```

Comment:
It's important to implement `equals` and `hashCode` that follow RFC. See how those are implemented in `IdentityContentCodec`.